builtin/providers/cloudstack/provider.go

```diff
@@ -1,30 +1,46 @@
 package cloudstack
 
 import (
+	"fmt"
+	"github.com/go-ini/ini"
 	"github.com/hashicorp/terraform/helper/schema"
 	"github.com/hashicorp/terraform/terraform"
+	"os/user"
 )
 
 // Provider returns a terraform.ResourceProvider.
 func Provider() terraform.ResourceProvider {
+	usr, _ := user.Current()
 	return &schema.Provider{
 		Schema: map[string]*schema.Schema{
+			"config": &schema.Schema{
+				Type:     schema.TypeString,
+				Optional: true,
+				Default:  fmt.Sprintf("%s/.cloudmonkey/config", usr.HomeDir),
+			},
+
+			"profile": &schema.Schema{
+				Type:     schema.TypeString,
+				Optional: true,
+				Default:  "",
+			},
+
 			"api_url": &schema.Schema{
 				Type:        schema.TypeString,
-				Required:    true,
-				DefaultFunc: schema.EnvDefaultFunc("CLOUDSTACK_API_URL", nil),
+				Optional:    true,
+				DefaultFunc: schema.EnvDefaultFunc("CLOUDSTACK_API_URL", ""),
```

Comment:
I thought the validation would fail if neither the environment variable nor the parameter was set with `nil`, but I tested and looks to be ok.